modules/light-clients/08-wasm/keeper/keeper_no_vm.go

```diff
@@ -0,0 +1,42 @@
+//go:build ibcwasm_novm 
+
+package keeper
+
+import (
+	storetypes "cosmossdk.io/core/store"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/internal/ibcwasm"
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+)
+
+// NewKeeperWithVM creates a new Keeper instance with the provided Wasm VM.
+// This constructor function is meant to be used when the chain uses x/wasm
+// and the same Wasm VM instance should be shared with it.
+func NewKeeperWithVM(
+	_ codec.BinaryCodec,
+	_ storetypes.KVStoreService,
+	_ types.ClientKeeper,
+	_ string,
+	_ ibcwasm.WasmEngine,
+	_ ibcwasm.QueryRouter,
+	_ ...Option,
+) Keeper {
+	panic("not implemented, please build with cgo enabled")
+}
+
+// NewKeeperWithConfig creates a new Keeper instance with the provided Wasm configuration.
+// This constructor function is meant to be used when the chain does not use x/wasm
+// and a Wasm VM needs to be instantiated using the provided parameters.
+func NewKeeperWithConfig(
+	_ codec.BinaryCodec,
+	_ storetypes.KVStoreService,
+	_ types.ClientKeeper,
+	_ string,
+	_ types.WasmConfig,
+	_ ibcwasm.QueryRouter,
+	_ ...Option,
+) Keeper {
+	panic("not implemented, please build with cgo enabled")
+}
```

Comment:
Agree with code rabbit, we should add some extra info in the godoc to state why this is

Reply:
I disagree with your suggest - it seems misleading and this module is not used in conjunction with x/wasm.

The purpose of this function is to be present when the binary is compiled with `ibcwasm_novm`, such that it will panic if used by an application developer when wasmvm dependencies are not present on the system.